### PR TITLE
Bugfix/ls24004467/dim with len

### PR DESCRIPTION
**Pull request by dom-apuliasoft**
## Description

Given a definition `$A` defined somewhere in the program, defining a definition `$B` using `DIM(%LEN($A))` should not throw any error and define `$B` accordingly.

From a technical standpoint a new branch was added in `CompileTimeInterpreter.evaluate` to properly address `LenExpr`s.

Related to:
- LS24004467

## Checklist:
- [x] If this feature involves RPGLE fixes or improvements, they are well-described in the summary.
- [x] There are tests for this feature.
- [x] RPGLE code used for tests is easily understandable and includes comments that clarify the purpose of this feature.
- [x] The code follows Kotlin conventions (run `./gradlew ktlintCheck`).
- [x] The code passes all tests (run `./gradlew check`).
- [ ] Relevant documentation is included in the `docs` directory.

### CODE REVIEW

File: rpgJavaInterpreter-core/src/test/resources/smeup/MUDRNRAPU00265.rpgle [hunk 1]
```diff
@@ -0,0 +1,13 @@
+      * Test declaration
+     D $SER            S              3  0 DIM(%LEN(SSERIE))
```
Comment (lanarimarco) [inline rpgJavaInterpreter-core/src/test/resources/smeup/MUDRNRAPU00265.rpgle:2]: What was the error?

File: rpgJavaInterpreter-core/src/test/resources/smeup/MUDRNRAPU00265.rpgle [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+     D STACK           DS                  OCCURS(300)
```
Comment (lanarimarco) [inline rpgJavaInterpreter-core/src/test/resources/smeup/MUDRNRAPU00265.rpgle:4]: Is this DS declaration necessary?

**Review by lanarimarco — CHANGES_REQUESTED** [review]

**Comment by dom-apuliasoft** [comment]
@lanarimarco I simplified the test case and moved the error description next to the erroring line.

File: rpgJavaInterpreter-core/src/test/kotlin/com/smeup/rpgparser/smeup/MULANGT02ConstAndDSpecTest.kt [hunk 3]
```diff
@@ -665,7 +665,17 @@ open class MULANGT02ConstAndDSpecTest : MULANGTTest() {
     }
 
     /**
-     * Assignment of integer value to a DS decimal subfield
+     * Like on a definition that is defined as a field of an occurable data structure
```
Comment (lanarimarco) [inline rpgJavaInterpreter-core/src/test/kotlin/com/smeup/rpgparser/smeup/MULANGT02ConstAndDSpecTest.kt:668]: I think that doc is not correct

File: rpgJavaInterpreter-core/src/test/kotlin/com/smeup/rpgparser/smeup/MULANGT02ConstAndDSpecTest.kt [hunk 4]
```diff
@@ -665,7 +665,17 @@ open class MULANGT02ConstAndDSpecTest : MULANGTTest() {
...
+     * @see #LS24004434
+     */
+    @Test
+    fun executeMUDRNRAPU00265() {
+        val expected = listOf(List(500) { "0" }.toString())
+        assertEquals(expected, "smeup/MUDRNRAPU00265".outputOf(configuration = smeupConfig))
+    }
+
+    /**
+     * Using %LEN on a definition inside a DIM
```
Comment (lanarimarco) [inline rpgJavaInterpreter-core/src/test/kotlin/com/smeup/rpgparser/smeup/MULANGT02ConstAndDSpecTest.kt:678]: Maybe this was the correct doc?

**Review by lanarimarco — CHANGES_REQUESTED** [review]
Doc issue

File: rpgJavaInterpreter-core/src/test/kotlin/com/smeup/rpgparser/smeup/MULANGT02ConstAndDSpecTest.kt (hunk 4 above)
Comment (dom-apuliasoft) [reply rpgJavaInterpreter-core/src/test/kotlin/com/smeup/rpgparser/smeup/MULANGT02ConstAndDSpecTest.kt]: Should be fixed now

**Review by lanarimarco — APPROVED** [review]

**Review by lanarimarco — APPROVED** [review]

**Pull request merged**